File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var checkSubarraySum = function(nums, k) {

    const seen = {[nums[0]%k]:0}
    let currSum = nums[0]
    for (let i = 1; i < nums.length; i++ ){
        currSum += nums[i]
        currSum %= k
        if (currSum == 0 || ((currSum in seen) && (i - seen[currSum]) > 1 )) return true
        else if (!(currSum in seen)) seen[currSum] = i
    }
    return false
    
};

// 5 2 4 0 1
// 5 1 5 5 0

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
from collections import defaultdict

class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        
        # x = n  * k
        
        # sum % k == 0
          #return true

        remain = defaultdict(int) 
        remain[0] = -1
        sum = 0
        for i,n in enumerate(nums): 
          #print(remain, sum)
              
          sum += n
          r = sum % k 

          if r not in remain: 
            remain[r] = i
            
          elif i - remain[r] > 1:
            return True

          
        return False
      
        
    